examples/evaluation_and_profiling/swe_bench/tests/test_iterative_predictor.py

```
@@ -0,0 +1,700 @@
+# =============================================================================
+# test_workspace_isolation - Concurrent instance isolation
+# =============================================================================
+
+class TestWorkspaceIsolation:
+    """Tests for workspace isolation between instances."""
+
+    def test_different_instance_ids_get_different_paths(self):
+        """Test that different instance_ids produce different workspace paths."""
+        from nat_swe_bench.predictors.predict_iterative.tools.git_tool import get_repo_path
+
+        workspace = "/tmp/workspace"
```

Ruff (0.14.14)

Probable insecure usage of temporary file or directory: "/tmp/workspace"

(S108)